addons/stock/models/stock_move_line.py

```diff
@@ -372,11 +372,15 @@ def write(self, vals):
 
         return res
 
-    def unlink(self):
-        precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+    @api.ondelete(at_uninstall=False)
+    def _check_move_line_state(self):
         for ml in self:
             if ml.state in ('done', 'cancel'):
                 raise UserError(_('You can not delete product moves if the picking is done. You can only correct the done quantities.'))
+
+    def unlink(self):
+        precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+        for ml in self:
```

Comment:
interesting, I missed the raise within the `try/except` block but so did the pylint checker